pkg/clusterconfig/clusterconfig.go

```diff
@@ -93,9 +92,6 @@ func Get() (*Config, error) {
 
 	cm, err := client.ConfigMaps(configNamespace).Get(configName, metaapi.GetOptions{})
 	if err != nil {
-		if errors.IsNotFound(err) {
-			return cfg, nil
-		}
```

Comment:
actually this was the key culprit, since you were always returning an empty configmap, the driver logic always fell through to the "unknown storage type" error.  But same result:  an error came back to the bootstrap logic and it doesn't handle it cleanly so it crashloops instead of creating the default resource.